### PR TITLE
Header

### PR DESCRIPTION
**Pull request by evertonfxavier**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "editor.wordWrap": "on",
+  "editor.wordWrapColumn": 60
+}
```
Comment (clovisdasilvaneto) [inline .vscode/settings.json:4]: Remove this guy and add it on .gitignore

File: src/components/Header/index.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import scrollTo from 'gatsby-plugin-smoothscroll';
+
+import Logo from '../Logo/';
+import Menu from '../Menu/';
+import SocialMedia from '../SocialMedia';
+import ScrollDown from '../ScrollDown/';
+import Title from '../../components/common/Title/';
+import Description from '../../components/common/Description/';
```
Comment (clovisdasilvaneto) [inline src/components/Header/index.jsx:9]: Don't need to be done on this PR, but please create an alias configuration in order to get rid of this relatives imports.

File: src/components/Header/index.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+import { HeaderWrapper, ContainerHeader, HeaderLogo, ContentHeader, BtnWrapper, BtnHeader, ScrollWrapper } from './styled';
+
```
Comment (clovisdasilvaneto) [inline src/components/Header/index.jsx:12]: Please create a constant in order to get rid of inline javascript functions, something like this 

```suggestion
const scrollToTarget = () => (id) => scrollTo(id)
```
and in your component, you can call it in this way:

```

          <BtnHeader onClick={scrollToTarget('#Header')}>
            start a project
          </BtnHeader>
```

Comment (clovisdasilvaneto) [reply src/components/Header/index.jsx:12]: This doesn't need to be done on this PR thought

File: src/components/Header/styled.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    &:hover{
+        background: ${({ theme }) => theme.colors.whiteDefault};
+        color: ${({ theme }) => theme.colors.writeHove};
+    }
+
+    ${({ theme }) => theme.media.phone`
+        margin-bottom: 14px;
+    `} 
+`
```
Comment (clovisdasilvaneto) [inline src/components/Header/styled.js:68]: This should be a separated component

File: src/components/Layout/index.jsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+
+import GlobalStyle from '../../utils/global/global.js';
+
+const Layout = ({ children }) => {
+  return (
+    <>
+      <GlobalStyle />
+        <main>{children}</main>
```
Comment (clovisdasilvaneto) [inline src/components/Layout/index.jsx:9]: Fix indentation.

File: src/components/Menu/styled.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+    font-family: ${({ theme }) => theme.fonts.montRegular};
+    color: ${({ theme }) => theme.colors.whiteDefault};
+    background-color: ${({ theme }) => theme.colors.bgMenu};
+    width: 280px;
+    height: 100%;
+    position: absolute;
+    right: 0;
+    z-index: 98;
+    position: fixed;
+    margin-right: -100%;
```
Comment (clovisdasilvaneto) [inline src/components/Menu/styled.js:49]: Please use `transform: translateX(280px)` in order to animate the sidebar, so in this way we will not have animation performance problem.

and on the **:checked** you will set `transform: translateX(0)` 👍

File: src/components/ScrollDown/index.jsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+import {ScrollStyled, LinkScroll, PositionLine, LineOne, LineTwo} from './styled';
+
+const ScrollDown = () => (
+  <ScrollStyled>
+
+    <LinkScroll href="./">Scroll Down</LinkScroll>
+
+    <PositionLine>
+      <LineOne></LineOne>
+      <LineTwo></LineTwo>
```
Comment (clovisdasilvaneto) [inline src/components/ScrollDown/index.jsx:12]: Please use `:before` & `:after`

File: src/components/SocialMedia/index.jsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        <LinkMedia href="https://twitter.com/" value="media">
+          <FigCapsule>
+            <TwitterWrapper />
+          </FigCapsule>
+          <TitleSocial>Twitter</TitleSocial>
+        </LinkMedia>
+      </LiMedia>
+
+      <LiMedia>
+        <LinkMedia href="https://www.instagram.com/" value="media">
```
Comment (clovisdasilvaneto) [inline src/components/SocialMedia/index.jsx:28]: This `value` properties does not exist, please remove it.

File: src/components/SocialMedia/styled.js [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+  top: 330px;
+  width: 100px;
+
+  ${({ theme }) => theme.media.phone`
+    display: none;
+  `}
+`
+
+export const UlMedia = styled.ul`
+
```
Comment (clovisdasilvaneto) [inline src/components/SocialMedia/styled.js:18]: remove extra line.

File: src/components/SocialMedia/styled.js [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+  &:hover {
+    span {
+      opacity: 1;
+      left: 0px;
+    }
+  }
+`
+
+export const FigCapsule = styled.figure`
+
```
Comment (clovisdasilvaneto) [inline src/components/SocialMedia/styled.js:45]: You don't need to create an empty element when you do not have styles attached on it. Please remove this element

File: src/utils/global/global.js [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+  body {
+    margin: 0;
+    padding: 0;
+  }
+
+  .container {
+    max-width: 1170px;
+    margin: 0 auto;
+    width: 100%;
+  }
```
Comment (clovisdasilvaneto) [inline src/utils/global/global.js:20]: Please create a component called `Container`

File: src/utils/theme/theme.js [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+const MAX_WIDTH = '1255px';
+
+const breakpoints = {
+  desktop: 960,
+  tablet: 768,
+  phone: 540,
+};
+
+const generateBreakpoint = (width, cssMarkup) => `
+  @media (max-width: ${width / RELATIVE_UNIT_BASE}em) {
```
Comment (clovisdasilvaneto) [inline src/utils/theme/theme.js:11]: I belive we can have some problems with this calculation on the future, please move it to: 

```suggestion
  @media (max-width: ${width}px) {
```

Comment (evertonfxavier) [reply src/utils/theme/theme.js]: Done.

File: src/utils/global/global.js (hunk 11 above)
Comment (evertonfxavier) [reply src/utils/global/global.js]: Done.

File: src/components/SocialMedia/styled.js (hunk 10 above)
Comment (evertonfxavier) [reply src/components/SocialMedia/styled.js]: Done.

File: src/components/SocialMedia/styled.js (hunk 9 above)
Comment (evertonfxavier) [reply src/components/SocialMedia/styled.js]: Done.

File: src/components/SocialMedia/index.jsx (hunk 8 above)
Comment (evertonfxavier) [reply src/components/SocialMedia/index.jsx]: Done.

File: src/components/Menu/styled.js (hunk 6 above)
Comment (evertonfxavier) [reply src/components/Menu/styled.js]: Done.

File: src/components/Layout/index.jsx (hunk 5 above)
Comment (evertonfxavier) [reply src/components/Layout/index.jsx]: Done.

File: src/components/Header/index.jsx (hunk 3 above)
Comment (evertonfxavier) [reply src/components/Header/index.jsx:12]: Done.

File: .vscode/settings.json (hunk 1 above)
Comment (evertonfxavier) [reply .vscode/settings.json]: Done.

File: src/components/Header/index.jsx (hunk 3 above)
Comment (evertonfxavier) [reply src/components/Header/index.jsx:12]: Done.

File: src/components/ScrollDown/index.jsx (hunk 7 above)
Comment (evertonfxavier) [reply src/components/ScrollDown/index.jsx]: Done!!!!! \õ/ \õ/ \õ/

**Pull request merged**